src/adc121c021/adc121c021.cxx

```diff
@@ -79,12 +79,20 @@ mraa_result_t ADC121C021::writeWord(uint8_t reg, uint16_t word)
 
 uint8_t ADC121C021::readByte(uint8_t reg)
 {
-  return mraa_i2c_read_byte_data(m_i2c, reg);
+  int val = mraa_i2c_read_byte_data(m_i2c, reg);
+  if (val != -1) {
+    return (uint8_t) val;
+  }
+  return 0;
```

Comment:
Agreed.  These older UPM drivers should be ported to use the C++ MRAA API at some point.